src/app/api/dataset-flow.api.ts

```diff
@@ -53,21 +54,35 @@ export class DatasetFlowApi {
     private datasetFlowsInitiatorsGQL = inject(DatasetFlowsInitiatorsGQL);
 
     public datasetTriggerFlow(params: {
+        accountId: string;
         datasetId: string;
         datasetFlowType: DatasetFlowType;
         flowRunConfiguration?: FlowRunConfiguration;
     }): Observable<DatasetTriggerFlowMutation> {
-        return this.datasetTriggerFlowGQL.mutate({ ...params }).pipe(
-            first(),
-            map((result: MutationResult<DatasetTriggerFlowMutation>) => {
-                /* istanbul ignore else */
-                if (result.data) {
-                    return result.data;
-                } else {
-                    throw new DatasetOperationError(result.errors ?? []);
-                }
-            }),
-        );
+        return this.datasetTriggerFlowGQL
+            .mutate(
+                { ...params },
+                {
+                    update: (cache) => {
+                        updateCacheHelper(cache, {
+                            accountId: params.accountId,
+                            datasetId: params.datasetId,
+                            fieldNames: ["data"],
```

Comment:
Why just `data`? Manual trigger of update or compacting/reset will most likely change the HEAD reference too, and that affects `metadata` as well. See we are resetting `metadata` when committing events in other places.

Reply:
removed